src/ChunkMap.h

```diff
@@ -460,15 +468,25 @@ class cChunkMap
 		/** Returns true if there is an entity with the specified ID within this layer's chunks */
 		bool HasEntity(UInt32 a_EntityID);
 
+		/** Returns how many chunks this layer has. */
+		size_t GetNumChunks();
+
 	protected:
 
 		cChunkPtr m_Chunks[LAYER_SIZE * LAYER_SIZE];
 		int m_LayerX;
 		int m_LayerZ;
 		cChunkMap * m_Parent;
-		int m_NumChunksLoaded;
 
 		cAllocationPool<cChunkData::sChunkSection> & m_Pool;
+	private:
+		/** Increase the internal chunk counter. */
+		void IncreaseChunkCounter();
+
+		/** Decrease the internal chunk counter. */
+		void DecreaseChunkCounter();
+
+		size_t m_ChunkCounter;
```

Comment:
And meanwhile, variables are strewn across various levels of indirection, each with their updater functions which need to be remembered to be called, and each with their own little bug when they're not after some large-ish change which neglected to do so?

This is what you said with regards to ParentChunk, IsTicking and IsChangingWorlds and is what I'm saying here.